From c1533eca436ac9f88e9fd2225b61c0e87cad0d69 Mon Sep 17 00:00:00 2001
From: Jason Rogena <jasonrogena@gmail.com>
Date: Sat, 7 Sep 2013 15:10:30 +0300
Subject: [PATCH 058/213] User now shown error message when he/she tries to
 enter combined milk production data when data for this date already exists or
 vice versa

---
 .idea/workspace.xml                    |   9 -
 web/log/debug.log                      | 389 +++++++++++++++++++++++++++++++++
 web/log/error.log                      |   2 +
 web/log/info.log                       |  55 +++++
 web/php/farmer/add_milk_production.php |  33 ++-
 5 files changed, 473 insertions(+), 15 deletions(-)

diff --git a/.idea/workspace.xml b/.idea/workspace.xml
index f9bed79..bd1ab29 100644
--- a/.idea/workspace.xml
+++ b/.idea/workspace.xml
@@ -469,17 +469,8 @@
       <change type="DELETED" beforePath="$PROJECT_DIR$/libraries/ActionBarSherlock/website/theming.html" afterPath="" />
       <change type="DELETED" beforePath="$PROJECT_DIR$/libraries/ActionBarSherlock/website/usage.html" afterPath="" />
       <change type="MODIFICATION" beforePath="$PROJECT_DIR$/libraries/ActionBarSherlock" afterPath="$PROJECT_DIR$/libraries/ActionBarSherlock" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/MilkProductionActivity.java" afterPath="$PROJECT_DIR$/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/MilkProductionActivity.java" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/MistroFarmer/build/source/r/debug/org/cgiar/ilri/mistro/farmer/R.java" afterPath="$PROJECT_DIR$/MistroFarmer/build/source/r/debug/org/cgiar/ilri/mistro/farmer/R.java" />
       <change type="MODIFICATION" beforePath="$PROJECT_DIR$/web/php/farmer/add_milk_production.php" afterPath="$PROJECT_DIR$/web/php/farmer/add_milk_production.php" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/web/php/common/database.php" afterPath="$PROJECT_DIR$/web/php/common/database.php" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/web/log/debug.log" afterPath="$PROJECT_DIR$/web/log/debug.log" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/web/log/error.log" afterPath="$PROJECT_DIR$/web/log/error.log" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/web/log/info.log" afterPath="$PROJECT_DIR$/web/log/info.log" />
       <change type="MODIFICATION" beforePath="$PROJECT_DIR$/libraries/ActionBarSherlock/actionbarsherlock/project.properties" afterPath="$PROJECT_DIR$/libraries/ActionBarSherlock/actionbarsherlock/project.properties" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/MistroFarmer/src/main/res/values/strings_en.xml" afterPath="$PROJECT_DIR$/MistroFarmer/src/main/res/values/strings_en.xml" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/MistroFarmer/build/res/all/debug/values/values.xml" afterPath="$PROJECT_DIR$/MistroFarmer/build/res/all/debug/values/values.xml" />
-      <change type="MODIFICATION" beforePath="$PROJECT_DIR$/.idea/workspace.xml" afterPath="$PROJECT_DIR$/.idea/workspace.xml" />
     </list>
     <ignored path="MistroFarmer.iws" />
     <ignored path=".idea/workspace.xml" />
diff --git a/web/log/debug.log b/web/log/debug.log
index d4fa990..af02d05 100644
--- a/web/log/debug.log
+++ b/web/log/debug.log
@@ -12744,3 +12744,392 @@
 )
 
 [Saturday, 7th September 2013 14:22:29] [fetch_milk_production_history.php]  cow milk production history encoded from an array and is '{"history":[{"id":"31","cow_id":"35","time":"2","quantity":"12","date_added":"2013-09-07 14:22:24","date":"2013-09-07","quantity_type":"KGs","name":"Rosy","ear_tag_number":"nai-rog-0"},{"id":"28","cow_id":"35","time":"1","quantity":"1","date_added":"2013-09-07 14:04:32","date":"2013-09-07","quantity_type":"KGs","name":"Rosy","ear_tag_number":"nai-rog-0"},{"id":"26","cow_id":"35","time":"0","quantity":"2","date_added":"2013-09-07 13:42:13","date":"2013-09-07","quantity_type":"KGs","name":"Rosy","ear_tag_number":"nai-rog-0"}]}'
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] json_decode returned: Array
+(
+    [fromID] => -1
+    [simCardSN] => 89254029541005994303
+)
+
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:00:18] [database.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:00:18] [database.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:00:18] [database.php] results gotten from 'SELECT `milk_production`.*,`cow`.`name`,`cow`.`ear_tag_number` FROM `farmer` INNER JOIN `cow` ON `farmer`.`id` = `cow`.`farmer_id` INNER JOIN `milk_production` ON `cow`.`id`=`milk_production`.`cow_id` WHERE `farmer`.`sim_card_sn`='89254029541005994303' ORDER BY `milk_production`.`id` DESC LIMIT 40' are Array
+(
+    [0] => Array
+        (
+            [id] => 26
+            [cow_id] => 35
+            [time] => 0
+            [quantity] => 2
+            [date_added] => 2013-09-07 13:42:13
+            [date] => 2013-09-07
+            [quantity_type] => KGs
+            [name] => Rosy
+            [ear_tag_number] => nai-rog-0
+        )
+
+)
+
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php]  cow milk production history encoded from an array and is '{"history":[{"id":"26","cow_id":"35","time":"0","quantity":"2","date_added":"2013-09-07 13:42:13","date":"2013-09-07","quantity_type":"KGs","name":"Rosy","ear_tag_number":"nai-rog-0"}]}'
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] json_decode returned: Array
+(
+    [cowEarTagNumber] => nai-rog-0
+    [quantity] => 12
+    [time] => 3
+    [quantityType] => KGs
+    [cowName] => Rosy
+    [simCardSN] => 89254029541005994303
+)
+
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:00:24] [database.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:00:24] [database.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:00:24] [database.php] results gotten from 'SELECT `id` FROM `cow` WHERE `ear_tag_number` = 'nai-rog-0' AND `name` = 'Rosy'' are Array
+(
+    [0] => Array
+        (
+            [id] => 35
+        )
+
+)
+
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] cow.id fetched and is 35
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] returning time '2013-09-07' using format 'Y-m-d'
+[Saturday, 7th September 2013 15:00:24] [database.php] results gotten from 'SELECT `time` FROM `milk_production` WHERE `cow_id` = 35 AND `date` = '2013-09-07'' are Array
+(
+    [0] => Array
+        (
+            [time] => 0
+        )
+
+)
+
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] json_decode returned: Array
+(
+    [cowEarTagNumber] => nai-rog-0
+    [quantity] => 12
+    [time] => 1
+    [quantityType] => KGs
+    [cowName] => Rosy
+    [simCardSN] => 89254029541005994303
+)
+
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:00:33] [database.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:00:33] [database.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:00:33] [database.php] results gotten from 'SELECT `id` FROM `cow` WHERE `ear_tag_number` = 'nai-rog-0' AND `name` = 'Rosy'' are Array
+(
+    [0] => Array
+        (
+            [id] => 35
+        )
+
+)
+
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] cow.id fetched and is 35
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] returning time '2013-09-07' using format 'Y-m-d'
+[Saturday, 7th September 2013 15:00:33] [database.php] results gotten from 'SELECT `time` FROM `milk_production` WHERE `cow_id` = 35 AND `date` = '2013-09-07'' are Array
+(
+    [0] => Array
+        (
+            [time] => 0
+        )
+
+)
+
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] returning time '2013-09-07 15:00:33' using format 'Y-m-d H:i:s'
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] json_decode returned: Array
+(
+    [cowEarTagNumber] => nai-rog-0
+    [quantity] => 12
+    [time] => 0
+    [quantityType] => KGs
+    [cowName] => Rosy
+    [simCardSN] => 89254029541005994303
+)
+
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:00:55] [database.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:00:55] [database.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:00:55] [database.php] results gotten from 'SELECT `id` FROM `cow` WHERE `ear_tag_number` = 'nai-rog-0' AND `name` = 'Rosy'' are Array
+(
+    [0] => Array
+        (
+            [id] => 35
+        )
+
+)
+
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] cow.id fetched and is 35
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] returning time '2013-09-07' using format 'Y-m-d'
+[Saturday, 7th September 2013 15:00:55] [database.php] results gotten from 'SELECT `time` FROM `milk_production` WHERE `cow_id` = 35 AND `date` = '2013-09-07'' are Array
+(
+    [0] => Array
+        (
+            [time] => 3
+        )
+
+)
+
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] json_decode returned: Array
+(
+    [fromID] => -1
+    [simCardSN] => 89254029541005994303
+)
+
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:01:00] [database.php] settings obtained: Array
+(
+    [db_name] => ilri_mistro
+    [mysql_creds] => Array
+        (
+            [username] => root
+            [password] => jason
+            [host] => localhost
+        )
+
+    [log_settings] => logs.ini
+    [response_codes] => response_codes.ini
+    [time_zone] => EAT
+)
+
+[Saturday, 7th September 2013 15:01:00] [database.php] response codes are: Array
+(
+    [db_error] => 32243
+    [user_not_authenticated] => 43322
+    [sim_card_registered] => 83242
+    [acknowledge_ok] => 533783
+    [data_error] => 934342
+    [no_data] => 77732
+)
+
+[Saturday, 7th September 2013 15:01:00] [database.php] results gotten from 'SELECT `milk_production`.*,`cow`.`name`,`cow`.`ear_tag_number` FROM `farmer` INNER JOIN `cow` ON `farmer`.`id` = `cow`.`farmer_id` INNER JOIN `milk_production` ON `cow`.`id`=`milk_production`.`cow_id` WHERE `farmer`.`sim_card_sn`='89254029541005994303' ORDER BY `milk_production`.`id` DESC LIMIT 40' are Array
+(
+    [0] => Array
+        (
+            [id] => 26
+            [cow_id] => 35
+            [time] => 3
+            [quantity] => 2
+            [date_added] => 2013-09-07 13:42:13
+            [date] => 2013-09-07
+            [quantity_type] => KGs
+            [name] => Rosy
+            [ear_tag_number] => nai-rog-0
+        )
+
+)
+
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php]  cow milk production history encoded from an array and is '{"history":[{"id":"26","cow_id":"35","time":"3","quantity":"2","date_added":"2013-09-07 13:42:13","date":"2013-09-07","quantity_type":"KGs","name":"Rosy","ear_tag_number":"nai-rog-0"}]}'
diff --git a/web/log/error.log b/web/log/error.log
index 3817cf5..96ec7ca 100644
--- a/web/log/error.log
+++ b/web/log/error.log
@@ -5,3 +5,5 @@
 [Saturday, 7th September 2013 14:04:23] [database.php] MySQL error 'Duplicate entry '35-0-2013-09-07' for key 'cow_id_2'' thrown while trying to run query 'INSERT INTO `milk_production`(`cow_id`,`time`,`quantity`, `quantity_type`,`date`,`date_added`) VALUES(35,0,1,'KGs','2013-09-07','2013-09-07 14:04:23')', exiting
 [Saturday, 7th September 2013 14:21:13] [database.php] MySQL error 'Duplicate entry '35-0-2013-09-07' for key 'cow_id_2'' thrown while trying to run query 'INSERT INTO `milk_production`(`cow_id`,`time`,`quantity`, `quantity_type`,`date`,`date_added`) VALUES(35,0,12,'KGs','2013-09-07','2013-09-07 14:21:13')', exiting
 [Saturday, 7th September 2013 14:22:21] [database.php] MySQL error 'Duplicate entry '35-1-2013-09-07' for key 'cow_id_2'' thrown while trying to run query 'INSERT INTO `milk_production`(`cow_id`,`time`,`quantity`, `quantity_type`,`date`,`date_added`) VALUES(35,1,12,'KGs','2013-09-07','2013-09-07 14:22:21')', exiting
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] a record already exists in the database with combined data for the date or the user is trying to enter combilned data when there is data record for this date, exiting
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] a record already exists in the database with combined data for the date or the user is trying to enter combilned data when there is data record for this date, exiting
diff --git a/web/log/info.log b/web/log/info.log
index c34cdab..de69820 100644
--- a/web/log/info.log
+++ b/web/log/info.log
@@ -1950,3 +1950,58 @@
 [Saturday, 7th September 2013 14:22:29] [database.php] running query SELECT `milk_production`.*,`cow`.`name`,`cow`.`ear_tag_number` FROM `farmer` INNER JOIN `cow` ON `farmer`.`id` = `cow`.`farmer_id` INNER JOIN `milk_production` ON `cow`.`id`=`milk_production`.`cow_id` WHERE `farmer`.`sim_card_sn`='89254029541005994303' ORDER BY `milk_production`.`id` DESC LIMIT 40
 [Saturday, 7th September 2013 14:22:29] [fetch_milk_production_history.php] sending milk production history data back to client as a json string
 [Saturday, 7th September 2013 14:22:29] [fetch_milk_production_history.php] gracefully exiting
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] Starting MilkProductionHistoryHandler
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] obtaining POST request
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:00:18] [database.php] Starting DatabaseHandler
+[Saturday, 7th September 2013 15:00:18] [database.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:00:18] [database.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:00:18] [database.php] connecting to ilri_mistro on localhost using root as the username
+[Saturday, 7th September 2013 15:00:18] [database.php] running query SELECT `milk_production`.*,`cow`.`name`,`cow`.`ear_tag_number` FROM `farmer` INNER JOIN `cow` ON `farmer`.`id` = `cow`.`farmer_id` INNER JOIN `milk_production` ON `cow`.`id`=`milk_production`.`cow_id` WHERE `farmer`.`sim_card_sn`='89254029541005994303' ORDER BY `milk_production`.`id` DESC LIMIT 40
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] sending milk production history data back to client as a json string
+[Saturday, 7th September 2013 15:00:18] [fetch_milk_production_history.php] gracefully exiting
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] Starting MilkProductionAdder
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] obtaining POST request
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:00:24] [add_milk_production.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:00:24] [database.php] Starting DatabaseHandler
+[Saturday, 7th September 2013 15:00:24] [database.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:00:24] [database.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:00:24] [database.php] connecting to ilri_mistro on localhost using root as the username
+[Saturday, 7th September 2013 15:00:24] [database.php] running query SELECT `id` FROM `cow` WHERE `ear_tag_number` = 'nai-rog-0' AND `name` = 'Rosy'
+[Saturday, 7th September 2013 15:00:24] [database.php] running query SELECT `time` FROM `milk_production` WHERE `cow_id` = 35 AND `date` = '2013-09-07'
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] Starting MilkProductionAdder
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] obtaining POST request
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:00:33] [database.php] Starting DatabaseHandler
+[Saturday, 7th September 2013 15:00:33] [database.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:00:33] [database.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:00:33] [database.php] connecting to ilri_mistro on localhost using root as the username
+[Saturday, 7th September 2013 15:00:33] [database.php] running query SELECT `id` FROM `cow` WHERE `ear_tag_number` = 'nai-rog-0' AND `name` = 'Rosy'
+[Saturday, 7th September 2013 15:00:33] [database.php] running query SELECT `time` FROM `milk_production` WHERE `cow_id` = 35 AND `date` = '2013-09-07'
+[Saturday, 7th September 2013 15:00:33] [database.php] running query INSERT INTO `milk_production`(`cow_id`,`time`,`quantity`, `quantity_type`,`date`,`date_added`) VALUES(35,1,12,'KGs','2013-09-07','2013-09-07 15:00:33')
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] returning response code 533783
+[Saturday, 7th September 2013 15:00:33] [add_milk_production.php] gracefully exiting
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] Starting MilkProductionAdder
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] obtaining POST request
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:00:55] [add_milk_production.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:00:55] [database.php] Starting DatabaseHandler
+[Saturday, 7th September 2013 15:00:55] [database.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:00:55] [database.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:00:55] [database.php] connecting to ilri_mistro on localhost using root as the username
+[Saturday, 7th September 2013 15:00:55] [database.php] running query SELECT `id` FROM `cow` WHERE `ear_tag_number` = 'nai-rog-0' AND `name` = 'Rosy'
+[Saturday, 7th September 2013 15:00:55] [database.php] running query SELECT `time` FROM `milk_production` WHERE `cow_id` = 35 AND `date` = '2013-09-07'
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] Starting MilkProductionHistoryHandler
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] obtaining POST request
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:01:00] [database.php] Starting DatabaseHandler
+[Saturday, 7th September 2013 15:01:00] [database.php] getting settings from: ../../config/settings.ini
+[Saturday, 7th September 2013 15:01:00] [database.php] getting response codes from: ../../config/response_codes.ini
+[Saturday, 7th September 2013 15:01:00] [database.php] connecting to ilri_mistro on localhost using root as the username
+[Saturday, 7th September 2013 15:01:00] [database.php] running query SELECT `milk_production`.*,`cow`.`name`,`cow`.`ear_tag_number` FROM `farmer` INNER JOIN `cow` ON `farmer`.`id` = `cow`.`farmer_id` INNER JOIN `milk_production` ON `cow`.`id`=`milk_production`.`cow_id` WHERE `farmer`.`sim_card_sn`='89254029541005994303' ORDER BY `milk_production`.`id` DESC LIMIT 40
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] sending milk production history data back to client as a json string
+[Saturday, 7th September 2013 15:01:00] [fetch_milk_production_history.php] gracefully exiting
diff --git a/web/php/farmer/add_milk_production.php b/web/php/farmer/add_milk_production.php
index ee601ef..5a2b21e 100644
--- a/web/php/farmer/add_milk_production.php
+++ b/web/php/farmer/add_milk_production.php
@@ -31,12 +31,33 @@ class MilkProductionAdder {
 			$cowID = $result[0]['id'];
 			$this->logHandler->log(4, $this->TAG,"cow.id fetched and is ".$cowID);
 			$dateEAT = $this->getTime('Y-m-d');
-			$timeEAT = $this->getTime('Y-m-d H:i:s');
-			$query = "INSERT INTO `milk_production`(`cow_id`,`time`,`quantity`, `quantity_type`,`date`,`date_added`) VALUES({$cowID},{$this->jsonObject['time']},{$this->jsonObject['quantity']},'{$this->jsonObject['quantityType']}','$dateEAT','$timeEAT')";
-			$this->database->runMySQLQuery($query, false, $this->codes['data_error']);
-			$this->logHandler->log(3, $this->TAG,"returning response code ".$this->codes['acknowledge_ok']);
-			echo $this->codes['acknowledge_ok'];
-			$this->logHandler->log(3, $this->TAG,"gracefully exiting");
+			
+			//check if there is a combined record for date
+			$query = "SELECT `time` FROM `milk_production` WHERE `cow_id` = {$cowID} AND `date` = '{$dateEAT}'";
+			$result = $this->database->runMySQLQuery($query, true);
+			$dataThereFlag = false;
+			if(sizeOf($result) > 0 && $this->jsonObject['time'] == 3) {
+				$dataThereFlag = true;
+			}
+			else {
+				for($i = 0; $i < sizeOf($result) ; $i++){
+					if($result[$i]["time"] == 3) {
+						$dataThereFlag = true;
+					}
+				}
+			}
+			if($dataThereFlag == false) {
+				$timeEAT = $this->getTime('Y-m-d H:i:s');
+				$query = "INSERT INTO `milk_production`(`cow_id`,`time`,`quantity`, `quantity_type`,`date`,`date_added`) VALUES({$cowID},{$this->jsonObject['time']},{$this->jsonObject['quantity']},'{$this->jsonObject['quantityType']}','$dateEAT','$timeEAT')";
+				$this->database->runMySQLQuery($query, false, $this->codes['data_error']);
+				$this->logHandler->log(3, $this->TAG,"returning response code ".$this->codes['acknowledge_ok']);
+				echo $this->codes['acknowledge_ok'];
+				$this->logHandler->log(3, $this->TAG,"gracefully exiting");
+			}
+			else {
+				$this->logHandler->log(1, $this->TAG,"a record already exists in the database with combined data for the date or the user is trying to enter combilned data when there is data record for this date, exiting");
+				die($this->codes['data_error']);
+			}
 		}
 		else {
 			$this->logHandler->log(1, $this->TAG,"it appears like there is no cow by the name '".$cowName."' and '".$earTagNumber."' as its ear tag number or more than one cow goes by these credentials, exiting");
-- 
2.7.1

